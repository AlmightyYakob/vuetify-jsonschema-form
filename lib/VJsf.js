// same as VJsjNoDeps but requiring each used component of vuetify to try working well with vuetify-loader

import VJsfNoNeps from './VJsfNoDeps'

import {
  VAvatar,
  VAutocomplete,
  VBtn,
  VCard,
  VCardActions,
  VCardText,
  VCardTitle,
  VCheckbox,
  VChip,
  VCombobox,
  VContainer,
  VDatePicker,
  VDialog,
  VExpansionPanels,
  VExpansionPanel,
  VExpansionPanelHeader,
  VExpansionPanelContent,
  VFlex,
  VIcon,
  VInput,
  VLayout,
  VListItemContent,
  VListItemTitle,
  VListTileContent,
  VListTileTitle,
  VMenu,
  VSelect,
  VSpacer,
  VSubheader,
  VTabs,
  VTab,
  VTabItem,
  VTextarea,
  VTextField,
  VTimePicker,
  VTooltip
} from 'vuetify/lib'

export default {
  name: 'VJsf',
  components: {
    VAvatar,
    VAutocomplete,
    VBtn,
    VCard,
    VCardActions,
    VCardText,
    VCardTitle,
    VCheckbox,
    VChip,
    VCombobox,
    VContainer,
    VDatePicker,
    VDialog,
    VExpansionPanels,
    VExpansionPanel,
    VExpansionPanelHeader,
    VExpansionPanelContent,
    VFlex,
    VIcon,
    VInput,
    VLayout,
    VListItemContent,
    VListItemTitle,
    VListTileContent,
    VListTileTitle,
    VMenu,
    VSelect,
    VSpacer,
    VSubheader,
    VTabs,
    VTab,
    VTabItem,
    VTextarea,
    VTextField,
    VTimePicker,
    VTooltip
  },
  mixins: [VJsfNoNeps]
}
