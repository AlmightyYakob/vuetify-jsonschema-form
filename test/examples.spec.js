import Vue from 'vue'
import { mount } from '@vue/test-utils'
import Vuetify from 'vuetify'
import VJsf from '../lib/index.vue'
const examples = require('../doc/examples')
Vue.use(Vuetify)

describe('Examples snapshot', () => {
  let vuetify

  beforeEach(() => {
    vuetify = new Vuetify({
      mocks: {
        $vuetify: {
          theme: { themes: {} }
        }
      }
    })
  })

  examples.forEach(example => {
    test(example.title, () => {
      const wrapper = mount(VJsf, { vuetify, propsData: example })
      expect(wrapper.isVueInstance()).toBeTruthy()
      expect(wrapper.element).toMatchSnapshot()
    })
  })
})
