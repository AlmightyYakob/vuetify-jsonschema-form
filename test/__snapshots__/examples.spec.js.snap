// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Examples snapshot Basic types 1`] = `
<div
  class="flex vjsf-property vjsf-property-root xs12 "
  description="All basic types are supported : string, number, integer, boolean"
  id="basic"
  title="Basic types"
/>
`;

exports[`Examples snapshot Dates 1`] = `
<div
  class="flex vjsf-property vjsf-property-root xs12 "
  description="The formats relative to date management are managed using the the date and time pickers from Vuetify."
  id="date-picker"
  template="
<p>BIM</p>
"
  title="Dates"
/>
`;
